kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -378,16 +406,35 @@ private synchronized void doApply(String deviceIdToBeConfigured, Map<String, Obj
         Optional<Device> device = getNetworkManagerDeviceByInterfaceId(deviceIdToBeConfigured);
         if (device.isPresent()) {
             if (configuredInterfaceIds.contains(deviceIdToBeConfigured)) {
-                manageConfiguredInterface(device.get(), deviceIdToBeConfigured, properties);
+                manageConfiguredInterface(device, deviceIdToBeConfigured, properties, Optional.empty());
             } else {
-                manageNonConfiguredInterface(device.get(), deviceIdToBeConfigured);
+                manageNonConfiguredInterface(device, deviceIdToBeConfigured);
+            }
+        } else {
+            if (isVlanType(deviceIdToBeConfigured, networkConfiguration)) {
+                manageConfiguredInterface(device, deviceIdToBeConfigured, properties, 
+                        Optional.of(NMDeviceType.NM_DEVICE_TYPE_VLAN));
             }
         }
     }
-
-    private synchronized void manageConfiguredInterface(Device device, String deviceId, NetworkProperties properties)
-            throws DBusException {
-        NMDeviceType deviceType = this.networkManager.getDeviceType(device.getObjectPath());
+    
+    private boolean isVlanType(String deviceId, Map<String, Object> networkConfiguration) {
+        Optional<NetInterfaceType> type = NetworkConfigurationServiceCommon.
+                getNetworkTypeFromProperties(deviceId, networkConfiguration);
+        return type.isPresent() && NetInterfaceType.VLAN.equals(type.get());
+    }
+    
+    private synchronized void manageConfiguredInterface(Optional<Device> device, String deviceId, NetworkProperties properties, 
+            Optional<NMDeviceType> knownDeviceType) throws DBusException {
+        NMDeviceType deviceType;
+        if (device.isPresent()) {
+            deviceType = this.networkManager.getDeviceType(device.get().getObjectPath());
+        } else if (knownDeviceType.isPresent()) {
+            deviceType = knownDeviceType.get();
+        } else {
+            logger.error("Unable to apply configuration to device {}: no device nor device type provided");
+            return;
+        }
```

Comment:
Following what was said in the previous comment:

```suggestion
    private synchronized void manageConfiguredInterface(Optional<Device> device, String deviceId, NetworkProperties properties) throws DBusException {
        NMDeviceType deviceType;
        if (device.isPresent()) {
            deviceType = this.networkManager.getDeviceType(device.get().getObjectPath());
        } else {
            deviceType = NMDeviceType.fromProperties(properties.getString("net.interface.%s.type", deviceId)); 
        ...
            // Again, handle the case in which the type is not set as you prefer
            logger.error("Unable to apply configuration to device {}: no device nor device type provided");
            return;
        }
```